package maidsafe.dht.kademlia.protobuf;

message Endpoint {
  required bytes ip = 1;
  required int32 port = 2;
}

message Contact {
  required bytes node_id = 1;
  optional Endpoint endpoint = 2;
  repeated bytes local_ips = 3;
  optional int32 local_port = 4;
  optional Endpoint rendezvous = 5;
  optional bool tcp443 = 6;
  optional bool tcp80 = 7;
  optional bytes public_key_id = 8;
  optional bytes public_key = 9;
  optional bytes other_info = 10;
}
